### ISSUE TITLE
[Rllib] Off policy estimation runs into assertionerrors during evaluation and crashes

### ISSUE THREAD

**Issue by jayanthnair** [ray-project/ray#37447]
### What happened + What you expected to happen

I am trying to test offline RL and specifically off policy estimation methods. I am using the tuned example data file (enormous.zip) for the pendulum environment as the training data. I generated an evaluation dataset using SAC locally and wanted to use this evaluation dataset in off policy evaluation.  However, I keep running into AssertionErrors like below during the evaluation stepv

```
# error

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
Cell In[5], line 62
     60 algo2 = config2.build()
     61 for i in range(300):
---> 62     algo2.train()
     64     if i % 10 == 0:
     65         checkpoint_dir = algo2.save()

File [c:\Users\jAYANTH.NAIR\Miniconda3\envs\raynightly\lib\site-packages\ray\tune\trainable\trainable.py:375](file:///C:/Users/jAYANTH.NAIR/Miniconda3/envs/raynightly/lib/site-packages/ray/tune/trainable/trainable.py:375), in Trainable.train(self)
    373 except Exception as e:
    374     skipped = skip_exceptions(e)
--> 375     raise skipped from exception_cause(skipped)
    377 assert isinstance(result, dict), "step() needs to return a dict."
    379 # We do not modify internal state nor update this result if duplicate.

File [c:\Users\jAYANTH.NAIR\Miniconda3\envs\raynightly\lib\site-packages\ray\tune\trainable\trainable.py:372](file:///C:/Users/jAYANTH.NAIR/Miniconda3/envs/raynightly/lib/site-packages/ray/tune/trainable/trainable.py:372), in Trainable.train(self)
    370 start = time.time()
    371 try:
--> 372     result = self.step()
    373 except Exception as e:
    374     skipped = skip_exceptions(e)

File [c:\Users\jAYANTH.NAIR\Miniconda3\envs\raynightly\lib\site-packages\ray\rllib\algorithms\algorithm.py:855](file:///C:/Users/jAYANTH.NAIR/Miniconda3/envs/raynightly/lib/site-packages/ray/rllib/algorithms/algorithm.py:855), in Algorithm.step(self)
    853 # Sequential: Train (already done above), then evaluate.
    854 if evaluate_this_iter and not self.config.evaluation_parallel_to_training:
--> 855     results.update(self._run_one_evaluation(train_future=None))
    857 # Attach latest available evaluation results to train results,
    858 # if necessary.
    859 if not evaluate_this_iter and self.config.always_attach_evaluation_results:

File [c:\Users\jAYANTH.NAIR\Miniconda3\envs\raynightly\lib\site-packages\ray\rllib\algorithms\algorithm.py:2880](file:///C:/Users/jAYANTH.NAIR/Miniconda3/envs/raynightly/lib/site-packages/ray/rllib/algorithms/algorithm.py:2880), in Algorithm._run_one_evaluation(self, train_future)
   2869     eval_results = eval_func_to_use(
   2870         duration_fn=functools.partial(
   2871             self._automatic_evaluation_duration_fn,
   (...)
   2876         )
   2877     )
   2878 # Run `self.evaluate()` only once per training iteration.
   2879 else:
-> 2880     eval_results = eval_func_to_use()
   2882 if self.evaluation_workers is not None:
   2883     # After evaluation, do a round of health check to see if any of
   2884     # the failed workers are back.
   2885     self.restore_workers(self.evaluation_workers)

File [c:\Users\jAYANTH.NAIR\Miniconda3\envs\raynightly\lib\site-packages\ray\rllib\algorithms\algorithm.py:1086](file:///C:/Users/jAYANTH.NAIR/Miniconda3/envs/raynightly/lib/site-packages/ray/rllib/algorithms/algorithm.py:1086), in Algorithm.evaluate(self, duration_fn)
   1084         ma_batch = ma_batch.as_multi_agent()
   1085         for batch in ma_batch.policy_batches.values():
-> 1086             assert batch.is_terminated_or_truncated()
   1087 # n timesteps per returned batch.
   1088 else:
   1089     num_units_done += (
   1090         _agent_steps
   1091         if self.config.count_steps_by == "agent_steps"
   1092         else _env_steps
   1093     )

AssertionError:

```

For reference, this is a sample of the evaluation json file (obtained using the batch builder code as given here: https://docs.ray.io/en/latest/rllib/rllib-offline.html )

```
# evaluation json file sample

{"type": "SampleBatch", "t": [0], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDQ/8m+7Ttrv2GjAb+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.24761474132537842]], "action_prob": [0.6759821176528931], "action_logp": [0.0], "rewards": [-3.9318583011627197], "prev_actions": [[0.0]], "prev_rewards": [0], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQB7cee+CVpkv1CKor+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [300]}
{"type": "SampleBatch", "t": [1], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQB7cee+CVpkv1CKor+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.6363822221755981]], "action_prob": [1.0767356157302856], "action_logp": [0.0], "rewards": [-4.3238372802734375], "prev_actions": [[-0.24761474132537842]], "prev_rewards": [-3.9318583011627197], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQBBVgu/LsJWv/xNCMCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [301]}
{"type": "SampleBatch", "t": [2], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQBBVgu/LsJWv/xNCMCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.8679929375648499]], "action_prob": [0.6099355220794678], "action_logp": [0.0], "rewards": [-5.063345432281494], "prev_actions": [[-0.6363822221755981]], "prev_rewards": [-4.3238372802734375], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDtDCq/SFw/v8A8QcCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [302]}
{"type": "SampleBatch", "t": [3], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDtDCq/SFw/v8A8QcCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.7282626628875732]], "action_prob": [1.644614577293396], "action_logp": [0.0], "rewards": [-6.191342830657959], "prev_actions": [[-0.8679929375648499]], "prev_rewards": [-5.063345432281494], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQBGHku/LtEbv5wZc8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [303]}
{"type": "SampleBatch", "t": [4], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQBGHku/LtEbv5wZc8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.6958649158477783]], "action_prob": [1.5891392230987549], "action_logp": [0.0], "rewards": [-7.6310200691223145], "prev_actions": [[-0.7282626628875732]], "prev_rewards": [-6.191342830657959], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCEkWi/cP3VvpPWjsCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [304]}
{"type": "SampleBatch", "t": [5], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCEkWi/cP3VvpPWjsCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.13653558492660522]], "action_prob": [0.16966770589351654], "action_logp": [0.0], "rewards": [-9.338830947875977], "prev_actions": [[-0.6958649158477783]], "prev_rewards": [-7.6310200691223145], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAmYXu/i6xBvgEumsCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [305]}
{"type": "SampleBatch", "t": [6], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAmYXu/i6xBvgEumsCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.6918716430664062]], "action_prob": [1.1072487831115723], "action_logp": [0.0], "rewards": [-11.03357982635498], "prev_actions": [[-0.13653558492660522]], "prev_rewards": [-9.338830947875977], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAdaH+/D1qLPWRcpcCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [306]}
{"type": "SampleBatch", "t": [7], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAdaH+/D1qLPWRcpcCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9390691518783569]], "action_prob": [4.944096565246582], "action_logp": [0.0], "rewards": [-12.120246887207031], "prev_actions": [[-0.6918716430664062]], "prev_rewards": [-11.03357982635498], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDSg3G/YcipPjG+rMCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [307]}
{"type": "SampleBatch", "t": [8], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDSg3G/YcipPjG+rMCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.7107090950012207]], "action_prob": [0.6675814986228943], "action_logp": [0.0], "rewards": [-10.776191711425781], "prev_actions": [[-0.9390691518783569]], "prev_rewards": [-12.120246887207031], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDgZFK/9NcRP3Cbq8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [308]}
{"type": "SampleBatch", "t": [9], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDgZFK/9NcRP3Cbq8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9893288612365723]], "action_prob": [8.661291122436523], "action_logp": [0.0], "rewards": [-9.308303833007812], "prev_actions": [[-0.7107090950012207]], "prev_rewards": [-10.776191711425781], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQC1gyW/D0xDP5Fup8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [309]}
{"type": "SampleBatch", "t": [10], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQC1gyW/D0xDP5Fup8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.987739622592926]], "action_prob": [10.750772476196289], "action_logp": [0.0], "rewards": [-7.911881446838379], "prev_actions": [[-0.9893288612365723]], "prev_rewards": [-9.308303833007812], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQA4G+G+Eu5lP+iansCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [310]}
{"type": "SampleBatch", "t": [11], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQA4G+G+Eu5lP+iansCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.8554260730743408]], "action_prob": [1.9000388383865356], "action_logp": [0.0], "rewards": [-6.564272403717041], "prev_actions": [[-0.987739622592926]], "prev_rewards": [-7.911881446838379], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQA+tGe+P1x5P+JCkcCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [311]}
{"type": "SampleBatch", "t": [12], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQA+tGe+P1x5P+JCkcCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.967752993106842]], "action_prob": [9.300251960754395], "action_logp": [0.0], "rewards": [-5.300942897796631], "prev_actions": [[-0.8554260730743408]], "prev_rewards": [-6.564272403717041], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCWx768Ou5/P5Usg8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [312]}
{"type": "SampleBatch", "t": [13], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCWx768Ou5/P5Usg8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9021782875061035]], "action_prob": [4.2699995040893555], "action_logp": [0.0], "rewards": [-4.224706649780273], "prev_actions": [[-0.967752993106842]], "prev_rewards": [-5.300942897796631], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAJ1CA+ptJ8P+OuZ8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [313]}
{"type": "SampleBatch", "t": [14], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAJ1CA+ptJ8P+OuZ8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9863784313201904]], "action_prob": [3.544543743133545], "action_logp": [0.0], "rewards": [-3.3111751079559326], "prev_actions": [[-0.9021782875061035]], "prev_rewards": [-4.224706649780273], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAWWJ8+XklzP6M3S8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [314]}
{"type": "SampleBatch", "t": [15], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAWWJ8+XklzP6M3S8CUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.8986021280288696]], "action_prob": [4.388782501220703], "action_logp": [0.0], "rewards": [-2.5847854614257812], "prev_actions": [[-0.9863784313201904]], "prev_rewards": [-3.3111751079559326], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQATH+A+nitmP7HaLsCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [315]}
{"type": "SampleBatch", "t": [16], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQATH+A+nitmP7HaLsCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9011911749839783]], "action_prob": [4.483258247375488], "action_logp": [0.0], "rewards": [-1.9989736080169678], "prev_actions": [[-0.8986021280288696]], "prev_rewards": [-2.5847854614257812], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQD/CAo/+JhXPwwAFcCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [316]}
{"type": "SampleBatch", "t": [17], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQD/CAo/+JhXPwwAFcCUjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9717451333999634]], "action_prob": [7.688287734985352], "action_logp": [0.0], "rewards": [-1.5484161376953125], "prev_actions": [[-0.9011911749839783]], "prev_rewards": [-1.9989736080169678], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDEvx4/x9VIP193/r+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [317]}
{"type": "SampleBatch", "t": [18], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDEvx4/x9VIP193/r+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9409891366958618]], "action_prob": [6.904288291931152], "action_logp": [0.0], "rewards": [-1.212202787399292], "prev_actions": [[-0.9717451333999634]], "prev_rewards": [-1.5484161376953125], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDFDi8/cso6P4FJ17+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [318]}
{"type": "SampleBatch", "t": [19], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDFDi8/cso6P4FJ17+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9890726804733276]], "action_prob": [5.609895706176758], "action_logp": [0.0], "rewards": [-0.9556190371513367], "prev_actions": [[-0.9409891366958618]], "prev_rewards": [-1.212202787399292], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCF9zs/VsstP5I4t7+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [319]}
{"type": "SampleBatch", "t": [20], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCF9zs/VsstP5I4t7+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.8728746175765991]], "action_prob": [2.7921457290649414], "action_logp": [0.0], "rewards": [-0.7648186683654785], "prev_actions": [[-0.9890726804733276]], "prev_rewards": [-0.9556190371513367], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDP60U/GV4iPwaRl7+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [320]}
{"type": "SampleBatch", "t": [21], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDP60U/GV4iPwaRl7+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9062695503234863]], "action_prob": [4.106199741363525], "action_logp": [0.0], "rewards": [-0.6155160665512085], "prev_actions": [[-0.8728746175765991]], "prev_rewards": [-0.7648186683654785], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCFo00/q3kYP3b1er+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [321]}
{"type": "SampleBatch", "t": [22], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCFo00/q3kYP3b1er+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.5504000186920166]], "action_prob": [0.138448566198349], "action_logp": [0.0], "rewards": [-0.5043822526931763], "prev_actions": [[-0.9062695503234863]], "prev_rewards": [-0.6155160665512085], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDO1lI/EDMRP4PfMr+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [322]}
{"type": "SampleBatch", "t": [23], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDO1lI/EDMRP4PfMr+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.8748334050178528]], "action_prob": [2.1528749465942383], "action_logp": [0.0], "rewards": [-0.4155934453010559], "prev_actions": [[-0.5504000186920166]], "prev_rewards": [-0.5043822526931763], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAap1Y/9X8LPyQpCb+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [323]}
{"type": "SampleBatch", "t": [24], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAap1Y/9X8LPyQpCb+UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9960087537765503]], "action_prob": [7.993715763092041], "action_logp": [0.0], "rewards": [-0.364790678024292], "prev_actions": [[-0.8748334050178528]], "prev_rewards": [-0.4155934453010559], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAQk1k/x+UGPwEP2r6UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [324]}
{"type": "SampleBatch", "t": [25], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAQk1k/x+UGPwEP2r6UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9553346037864685]], "action_prob": [8.198713302612305], "action_logp": [0.0], "rewards": [-0.32981476187705994], "prev_actions": [[-0.9960087537765503]], "prev_rewards": [-0.364790678024292], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDkr1s/1W0DP6Bzor6UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [325]}
{"type": "SampleBatch", "t": [26], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDkr1s/1W0DP6Bzor6UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.922048807144165]], "action_prob": [3.7423226833343506], "action_logp": [0.0], "rewards": [-0.30413544178009033], "prev_actions": [[-0.9553346037864685]], "prev_rewards": [-0.32981476187705994], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAtDF0/tSABP57eVb6UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [326]}
{"type": "SampleBatch", "t": [27], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQAtDF0/tSABP57eVb6UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9758843779563904]], "action_prob": [14.289471626281738], "action_logp": [0.0], "rewards": [-0.28768789768218994], "prev_actions": [[-0.922048807144165]], "prev_rewards": [-0.30413544178009033], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDs1l0/Uob/PliO/L2UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [327]}
{"type": "SampleBatch", "t": [28], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQDs1l0/Uob/PliO/L2UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9846287965774536]], "action_prob": [19.06915283203125], "action_logp": [0.0], "rewards": [-0.27843335270881653], "prev_actions": [[-0.9758843779563904]], "prev_rewards": [-0.28768789768218994], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQBmH14//on+PjLgNb2UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [328]}
{"type": "SampleBatch", "t": [29], "eps_id": [1726495364229716], "agent_index": [0], "obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQBmH14//on+PjLgNb2UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "actions": [[-0.9462288022041321]], "action_prob": [5.693748950958252], "action_logp": [0.0], "rewards": [-0.274497926235199], "prev_actions": [[-0.9846287965774536]], "prev_rewards": [-0.27843335270881653], "terminateds": [false], "truncateds": [false], "infos": [{}], "new_obs": "BCJNGGhAjAAAAAAAAABOhwAAAFKABZWBAAEA8hmMEm51bXB5LmNvcmUubnVtZXJpY5SMC19mcm9tYnVmZmVylJOUKJYMLgDxAQCf1l0/X4f/PuihNj2UjAU8APEWlIwFZHR5cGWUk5SMAmY0lImIh5RSlChLA4wBPJROTk5K/////wUA8AVLAHSUYksBSwOGlIwBQ5R0lFKULgAAAAA=", "unroll_id": [329]}



```

Irrespective of whether the last step says true or false for terminated/truncated, I see the same error. 

### Versions / Dependencies

OS - Windows
Using a ray nightly build due to https://github.com/ray-project/ray/issues/37042#issuecomment-1622042085
```
# requirements.txt
absl-py==1.4.0
adal==1.2.7
aiohttp==3.8.4
aiohttp-cors==0.7.0
aiorwlock==1.3.0
aiosignal==1.3.1
ansicon==1.89.0
anyio==3.7.0
argcomplete==2.1.2
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
arrow==1.2.3
asttokens==2.2.1
async-timeout==4.0.2        
attrs==23.1.0
azure-common==1.1.28
azure-core==1.27.1
azure-graphrbac==0.61.1
azure-identity==1.13.0
azure-mgmt-authorization==3.0.0
azure-mgmt-containerregistry==10.1.0
azure-mgmt-core==1.4.0
azure-mgmt-keyvault==10.2.2
azure-mgmt-resource==21.2.1
azure-mgmt-storage==20.1.0
azure-storage-blob==12.13.0
azureml-core==1.48.0
azureml-dataprep==4.8.6
azureml-dataprep-native==38.0.0
azureml-dataprep-rslex==2.15.2
azureml-dataset-runtime==1.48.0
azureml-defaults==1.48.0
azureml-inference-server-http==0.7.7
azureml-mlflow==1.52.0
backcall==0.2.0
backports.tempfile==1.0
backports.weakref==1.0.post1
bcrypt==4.0.1
beautifulsoup4==4.12.2
bleach==6.0.0
blessed==1.20.0
blinker==1.6.2
cachetools==5.3.1
certifi @ file:///C:/b/abs_85o_6fm0se/croot/certifi_1671487778835/work/certifi
cffi==1.15.1
charset-normalizer==3.1.0
click==8.1.3
cloudpickle==2.2.1
colorama==0.4.6
colorful==0.5.5
comm==0.1.3
contextlib2==21.6.0
contourpy==1.1.0
cryptography==38.0.4
cycler==0.11.0
databricks-cli==0.17.7
debugpy==1.6.7
decorator==5.1.1
defusedxml==0.7.1
distlib==0.3.6
distro==1.8.0
dm-tree==0.1.8
docker==6.1.3
dotnetcore2==3.1.23
entrypoints==0.4
exceptiongroup==1.1.2
executing==1.2.0
fastapi==0.99.1
fastjsonschema==2.17.1
filelock==3.12.2
Flask==2.3.2
Flask-Cors==3.0.10
fonttools==4.40.0
fqdn==1.5.1
frozenlist==1.3.3
fsspec==2023.6.0
fusepy==3.0.1
gast==0.5.4
gitdb==4.0.10
GitPython==3.1.31
google-api-core==2.11.1
google-auth==2.21.0
google-auth-oauthlib==1.0.0
googleapis-common-protos==1.59.1
gpustat==1.1
grpcio==1.56.0
Gymnasium==0.26.3
gymnasium-notices==0.0.1
h11==0.14.0
humanfriendly==10.0
idna==3.4
imageio==2.31.1
importlib-metadata==6.7.0
inference-schema==1.5.1
ipykernel==6.24.0
ipython==8.14.0
ipython-genutils==0.2.0
ipywidgets==8.0.6
isodate==0.6.1
isoduration==20.11.0
itsdangerous==2.1.2
jedi==0.18.2
jeepney==0.8.0
Jinja2==3.1.2
jinxed==1.2.0
jmespath==1.0.1
joblib==1.3.1
jsonpickle==2.2.0
jsonpointer==2.4
jsonschema==4.17.3
jupyter==1.0.0
jupyter-console==6.6.3
jupyter-events==0.6.3
jupyter_client==8.3.0
jupyter_core==5.3.1
jupyter_server==2.7.0
jupyter_server_terminals==0.4.4
jupyterlab-pygments==0.2.2
jupyterlab-widgets==3.0.7
kiwisolver==1.4.4
knack==0.10.1
lazy_loader==0.3
lz4==4.3.2
Markdown==3.4.3
markdown-it-py==3.0.0
MarkupSafe==2.1.3
matplotlib==3.7.2
matplotlib-inline==0.1.6
mdurl==0.1.2
mistune==3.0.1
mlflow-skinny==2.4.1
mpmath==1.3.0
msal==1.22.0
msal-extensions==1.0.0
msgpack==1.0.5
msrest==0.7.1
msrestazure==0.6.4
multidict==6.0.4
nbclassic==1.0.0
nbclient==0.8.0
nbconvert==7.6.0
nbformat==5.9.0
ndg-httpsclient==0.5.1
nest-asyncio==1.5.6
networkx==3.1
notebook==6.5.4
notebook_shim==0.2.3
numpy==1.24.3
nvidia-ml-py==11.525.131
oauthlib==3.2.2
opencensus==0.11.2
opencensus-context==0.1.3
opencensus-ext-azure==1.1.9
overrides==7.3.1
packaging==21.3
pandas==2.0.2
pandocfilters==1.5.0
paramiko==2.12.0
parso==0.8.3
pathspec==0.11.1
pickleshare==0.7.5
Pillow==10.0.0
pkginfo==1.9.6
platformdirs==3.8.0
portalocker==2.7.0
prometheus-client==0.17.0
prompt-toolkit==3.0.38
protobuf==4.23.3
psutil==5.8.0
pure-eval==0.2.2
py-spy==0.3.14
pyarrow==6.0.1
pyasn1==0.5.0
pyasn1-modules==0.3.0
pycparser==2.21
pydantic==1.10.10
Pygments==2.15.1
PyJWT==2.7.0
PyNaCl==1.5.0
pyOpenSSL==22.1.0
pyparsing==3.0.9
pyreadline3==3.4.1
pyrsistent==0.19.3
PySocks==1.7.1
python-dateutil==2.8.2
python-json-logger==2.0.7
pytz==2023.3
PyWavelets==1.4.1
pywin32==306
pywinpty==2.0.10
PyYAML==6.0
pyzmq==25.1.0
qtconsole==5.4.3
QtPy==2.3.1
ray @ https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-3.0.0.dev0-cp310-cp310-win_amd64.whl
ray-on-aml==0.2.4
requests==2.31.0
requests-oauthlib==1.3.1
rfc3339-validator==0.1.4
rfc3986-validator==0.1.1
rich==13.4.2
rsa==4.9
scikit-image==0.21.0
scikit-learn==1.3.0
scipy==1.10.1
SecretStorage==3.3.3
Send2Trash==1.8.2
six==1.16.0
smart-open==6.3.0
smmap==5.0.0
sniffio==1.3.0
soupsieve==2.4.1
sqlparse==0.4.4
stack-data==0.6.2
starlette==0.27.0
sympy==1.12
tabulate==0.9.0
tensorboard==2.13.0
tensorboard-data-server==0.7.1
tensorboardX==2.6.1
tensorflow-probability==0.20.1
terminado==0.17.1
threadpoolctl==3.1.0
tifffile==2023.4.12
tinycss2==1.2.1
torch==2.0.1
tornado==6.3.2
traitlets==5.9.0
typer==0.9.0
typing_extensions==4.7.1
tzdata==2023.3
uri-template==1.3.0
urllib3==1.26.16
uvicorn==0.22.0
virtualenv==20.21.0
waitress==2.1.2
wcwidth==0.2.6
webcolors==1.13
webencodings==0.5.1
websocket-client==1.6.1
Werkzeug==2.3.6
widgetsnbextension==4.0.7
wincertstore==0.2
wrapt==1.12.1
yarl==1.9.2
zipp==3.15.0
```

### Reproduction script

```
from ray.rllib.algorithms import cql as cql
from ray.rllib.utils.framework import try_import_torch

from ray.rllib.offline.estimators import (
    ImportanceSampling,
    WeightedImportanceSampling,
    DirectMethod,
    DoublyRobust,
)
torch, _ = try_import_torch()

if __name__ == "__main__":

        
    config2 = (
        cql.CQLConfig()
        .debugging(log_level="INFO")
        .framework(framework="torch")
        .rollouts(num_rollout_workers=0,
                  num_envs_per_worker= 20,)
                  #enable_connectors= True,   
                  #observation_filter= "MeanStdFilter",)

        .reporting(min_train_timesteps_per_iteration=1000)
        .environment(normalize_actions=True, env="Pendulum-v1")
        .offline_data(input_ = "./enormous.json", 
            #input_ = "data/Pendulum_example",

                      actions_in_input_normalized= True)

        .evaluation(
            evaluation_num_workers=1,
            evaluation_interval=3,
            evaluation_duration=10,
            evaluation_duration_unit="episodes",
            evaluation_parallel_to_training=False,
            evaluation_config={"input": "./demo-out/output-2023-07-14_18-06-05_worker-0_0.json","explore": False}, 
            off_policy_estimation_methods={

                "wis": {"type": WeightedImportanceSampling},

            
            },

        )
    )      
        
    algo2 = config2.build()
    for i in range(300):
        algo2.train()

        if i % 10 == 0:
            checkpoint_dir = algo2.save()
            print(f"Checkpoint saved in directory {checkpoint_dir}")
        
# Get policy and model.
cql_policy = algo2.get_policy()
cql_model = cql_policy.model


```

### Issue Severity

High: It blocks me from completing my task.

**Comment by ArturNiederfahrenhorst** [comment]
Hi @jayanthnair ,

This is not a bug, but the error message could better, admittedly. (I'm opening a PR for that)

`evaluation_duration_unit="episodes",
Makes it so that RLlib requires the evaluation data to be presented in terminated or truncated format.
You SAC batches don't seem to satisfy this requirement.

If you change to source dataset to `ray/rllib/tests/data/pendulum/large.json`, it will run fine.
Note that large.json is comprised of episodic data, so this only shows that the rest of your script is fine.
It's just your evaluation data not being episodic.

Feel free to repoen this issue if you think there is a problem that needs solving from our side!

**Comment by JulianKlug** [comment]
Alternative solution, set `evaluation_duration_unit` to `"timesteps"` instead of `"episodes"`:

> `evaluation_duration_unit="timesteps"`
